docs/FAQ.md

```diff
@@ -35,3 +60,54 @@ It first find the payment by `key` where `key=${merchantReference}` and then it
+- Both modules should be deployed as a publicly exposed services.
+- Modules are **stateless** which makes running multiple instances in parallel possible.
+- If the modules are deployed into a Kubernetes cluster, then it is recommended to **enable horizontal scaling** with at least 2 running instances at the same time in order to omit downtime possibility.
+
+### Is there any recommendation for securing the integration services as those need to be publicly exposed services ?
+
+- An encrypted HTTPS connection strongly recommended for production setups instead of HTTP connection.
+- To protect your extension service from unauthorized calls, we recommend to [activate basic auth authentication](../extension/docs/HowToRun.md#commercetools) on the API extension to prevent unauthorized 3rd-party from accessing extension endpoint.
+- To protect your notification service from unauthorized notifications, we strongly recommend that you activate Hash-based message authentication code [HMAC signatures](../notification/docs/IntegrationGuide.md#step-1-set-up-notification-webhook-and-generate-hmac-signature) during the notification setup.
+
+### What is the best practices for creating/maintaining commercetools payment object ?
+
+- Create your commercetools payment with following the [creating commercetools payment guide](../extension/docs/WebComponentsIntegrationGuide.md#step-2-creating-a-commercetools-payment).
+- After successful payment creation always [add](https://docs.commercetools.com/api/projects/carts#add-payment) it to the appropriate cart.
+- **Never delete or un-assign** created payment objects during the checkout from the cart.
+- Currently [payment status](https://docs.commercetools.com/api/projects/payments#paymentstatus) is not maintained by the integration as status of the payment can be derived from the state of the payment transaction(s).
+
+### Is there any error handling mechanism required while using commercetools-adyen-integration ?
+
+Yes, in case you encounter errors in your integration, refer to the [error-handling section](../extension/docs/WebComponentsIntegrationGuide.md#error-handling) on the integration documentation.
+
+### When we should create commercetools order ? 
+
+There are 2 approaches for creating commercetools order, one is creating order after a successful payment, and other is creating before the successful payment.
+Both approaches have their good and bad sides, but we found out that creating an order AFTER a successful payment is less harmful. Please read the details below for both approaches and decide based on your checkout scenarios.
+
+#### What to consider when creating order _AFTER_ a successful commercetools payment ?
+
+- **Shop was not reachable due to network issues**: It is possible that shop was not reachable due to network issues. In that case there will be a successful payment but no order.
```

Comment:
```suggestion
- **Shop (success  redirect URL) is not reachable due to the network issues**: Since shop creates an order and  success shop redirect URL can not be reached we might end up with a successful payment but no order.
```